src/components/Todo.tsx

```diff
@@ -0,0 +1,122 @@
+import cn from 'classnames';
+import { Todo } from '../types/Todo';
+import { useEffect, useRef, useState } from 'react';
+
+type Props = {
+  todo: Todo;
+  toggleTodo: (id: number) => void;
+  deletePost: (id: number) => void;
+  isLoading: boolean;
+  updatedPost: (todo: Todo) => void;
+};
+
+export const TodoComponent: React.FC<Props> = ({
+  todo,
+  toggleTodo,
+  deletePost,
+  isLoading,
+  updatedPost,
+}) => {
+  const { id, title, completed } = todo;
+
+  const [formTodo, setFormTodo] = useState(false);
+  const [inputText, setInputText] = useState(title);
+
+  const refInputUpdate = useRef<HTMLInputElement>(null);
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setInputText(e.target.value);
+  };
+
+  useEffect(() => {
+    if (refInputUpdate.current) {
+      refInputUpdate.current.focus();
+    }
+
+    const handleKeyDown = (e: KeyboardEvent) => {
+      if (e.key === 'Escape') {
+        setFormTodo(false);
+      }
+    };
+
+    document.addEventListener('keydown', handleKeyDown);
+
+    return () => {
+      document.removeEventListener('keydown', handleKeyDown);
+    };
+  }, [formTodo]);
+
+  return (
+    <div key={id} data-cy="Todo" className={cn('todo', { completed })}>
+      <label className="todo__status-label" htmlFor={`todo-status-${id}`}>
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          id={`todo-status-${id}`}
+          checked={completed}
+          onChange={() => toggleTodo(id)}
+        />
+        <span className="visually-hidden">Mark as completed</span>
+      </label>
+
+      {formTodo ? (
+        <form
+          onSubmit={() => {
+            if (inputText.length <= 0) {
+              deletePost(id);
+            } else {
+              updatedPost({ ...todo, title: inputText });
+            }
+
+            setFormTodo(false);
+          }}
+        >
+          <input
+            className="todo__title-field"
+            type="text"
+            value={inputText}
+            onChange={handleInputChange}
+            ref={refInputUpdate}
+            onBlur={() => {
```

Comment:
on blur, you should also send a request (behavior should be the same as on submit):
![image](https://github.com/user-attachments/assets/cac15729-0499-4e5a-8edf-a6b2769f9161)